docs/gallery/general/extensions.py

```diff
@@ -225,3 +225,93 @@ def __init__(self, **kwargs):
 
 io = NWBHDF5IO('cache_spec_example.nwb', mode='r', load_namespaces=True)
 nwbfile = io.read()
+
+####################
+# .. _MultiContainerInterface:
+# Creating and using a custom MultiContainerInterface
+# -----------------------------------------------------
+# It is sometimes the case that we need a group to hold zero-or-more or one-or-more of the same object.
+# Here we show how to create an extension that defines a group (`PotatoSack`) that holds multiple objects (`Pototo`es)
+# and then how to use the new data types. First, we use `pynwb` to define the new data types.
+
+from pynwb.spec import NWBNamespaceBuilder, NWBGroupSpec, NWBAttributeSpec
+
+name = 'test_multicontainerinterface'
+ns_path = name + ".namespace.yaml"
+ext_source = name + ".extensions.yaml"
+
+potato = NWBGroupSpec(neurodata_type_def='Potato',
+                      neurodata_type_inc='NWBDataInterface',
+                      doc='time of multicontainer', quantity='*',
+                      attributes=[
+                          NWBAttributeSpec(name='weight',
+                                           doc='weight of potato',
+                                           dtype='float',
+                                           required=True),
+                          NWBAttributeSpec(name='age',
+                                           doc='age of potato',
+                                           dtype='float',
+                                           required=False)
+                      ])
+
+potato_sack = NWBGroupSpec(neurodata_type_def='PotatoSack',
+                           neurodata_type_inc='NWBDataInterface',
+                           name='potato_sack',
+                           doc='test of multicontainer', quantity='?',
+                           groups=[potato],
+                           attributes=[
+                               NWBAttributeSpec(name='help',
+                                                doc='help',
+                                                dtype='text',
+                                                value="It's a sack of potatoes")
+                           ])
+
+ns_builder = NWBNamespaceBuilder(name + ' extensions', name)
+ns_builder.add_spec(ext_source, potato_sack)
+ns_builder.export(ns_path)
+
+####################
+# Then create DataMapper classes registered to the new data types (this is generally done in a new file)
```

Comment:
Your use of *DataMapper* here suggests this is a construct in the API. I'm afraid this will confuse users. Can you change the language ?

Reply:
Sure, should I just call it a class?

Reply:
*Container class* would be good. That's how we were refer to this layer in the software architecture overview: http://pynwb.readthedocs.io/en/latest/overview_software_architecture.html#main-concepts

Reply:
ok, will do